### PR TITLE
[roseus] Support private rosname in roseus

### PR DESCRIPTION
**Pull request by wkentaro**
Close https://github.com/jsk-ros-pkg/jsk_roseus/issues/229

### CODE REVIEW

**Comment by k-okada** [comment]
thanks, may be we need similar fix for service call?

--
◉ Kei Okada

**Comment by wkentaro** [comment]
Yeah, I think so.

File: roseus/roseus.cpp [hunk 1]
```diff
@@ -877,6 +877,7 @@ pointer ROSEUS_ADVERTISE(register context *ctx,int n,pointer *argv)
   ckarg2(2,4);
   if (isstring(argv[0])) topicname.assign((char *)get_string(argv[0]));
   else error(E_NOSTRING);
+  topicname = ros::names::resolve(topicname);
```
Comment (k-okada) [inline roseus/roseus.cpp]:
```
ros::names::resolve() resolves Graph Resource Names in ROS

Node            nRelative (default)      Global          Private
/node1          bar -> /bar             /bar -> /bar    ~bar -> /node1/bar
/wg/node2       bar -> /wg/bar          /bar -> /bar    ~bar -> /wg/node2/bar
/wg/node3       foo/bar -> /wg/foo/bar  /foo/bar -> /foo/bar    ~foo/bar -> /wg\
/node3/foo/bar

```

**Comment by wkentaro** [comment]
@k-okada Could u please merge this?

**Comment by wkentaro** [comment]
Please merge this.

**Comment by wkentaro** [comment]
Well, I will change the topicname and servicename for subscribe/advertiseService also.

**Comment by wkentaro** [comment]
I fixed the rosname for both topic and service communications. Could you please review?

File: roseus/test/test-talker-listener.l [hunk 2]
```diff
@@ -15,11 +15,15 @@
 ;; method call
 (defclass string-cb-class
   :super propertied-object
-  :slots ())
+  :slots (msg-camep))
```
Comment (k-okada) [inline roseus/test/test-talker-listener.l]: what is camep

Comment (k-okada) [inline roseus/test/test-talker-listener.l]: Oh, that is came-p, i see

File: roseus/roseus.cpp [hunk 3]
```diff
@@ -769,8 +769,12 @@ pointer ROSEUS_SUBSCRIBE(register context *ctx,int n,pointer *argv)
 
   // ;; arguments ;;
   // topicname message_type callbackfunc args0 ... argsN [ queuesize ] [ :groupname groupname ]
-  if (isstring(argv[0])) topicname.assign((char *)get_string(argv[0]));
-  else error(E_NOSTRING);
+  if (isstring(argv[0])) {
+    topicname.assign((char *)get_string(argv[0]));
+    topicname = ros::names::resolve(topicname);
```
Comment (k-okada) [inline roseus/roseus.cpp]: do we have error or warning is we use 
```
topicname = ros::names::resolve((char *)get_string(argv[0]))
```
or
```
ortopicname = ros::names::resolve(std::string(char *)get_string(argv[0])));
```
??

**Comment by k-okada** [comment]
> Well, I will change the topicname and servicename for subscribe/advertiseService also.

Thank you, I'm waiting for this updaets ;-)

File: roseus/roseus.cpp (hunk 3 above)
Comment (wkentaro) [inline roseus/roseus.cpp]: According to http://stackoverflow.com/a/3228982, it should works.

File: roseus/test/test-talker-listener.l (hunk 2 above)
Comment (wkentaro) [inline roseus/test/test-talker-listener.l]: should it be `msg-came-p`?

**Comment by wkentaro** [comment]
Fixed.

**Comment by k-okada** [comment]
Is 
```
topicname = ros::names::resolve(std::string(char *)get_string(argv[0])));
```
not working? it seems most of jobs failing on test, because of something like memory leaK???

**Comment by wkentaro** [comment]
> Is
topicname = ros::names::resolve(std::string(char *)get_string(argv[0])));
not working? it seems most of jobs failing on test, because of something like memory leaK???

I'm not sure why it does not work.

**Comment by wkentaro** [comment]
Now I found another trial to be executed.
I hope it works.

**Comment by k-okada** [comment]
rewrited at #478

**Comment by k-okada** [comment]
released as 1.6.0

**Pull request closed without merging**